patches/api/0421-Paper-Item-API.patch

```diff
@@ -0,0 +1,582 @@
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/inventory/item/properties/AttributePair.java
+@@ -0,0 +1,24 @@
++package io.papermc.paper.inventory.item.properties;
++
++import org.bukkit.attribute.Attribute;
++import org.bukkit.attribute.AttributeModifier;
++import org.jetbrains.annotations.NotNull;
++
++/**
++ * Represents a serialized attribute that is grouped with a modifier.
++ */
++public interface AttributePair {
++
++    /**
++     * Gets the attribute
++     * @return attribute
++     */
++    @NotNull Attribute attribute();
++
++    /**
++     * Gets the modifier
++     * @return modifier
++     */
++    @NotNull AttributeModifier modifier();
++
++}
+diff --git a/src/main/java/io/papermc/paper/inventory/item/properties/EnchantmentPair.java b/src/main/java/io/papermc/paper/inventory/item/properties/EnchantmentPair.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..9733d750bfc0bcabf995fe54ba8c9d2c180df85c
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/inventory/item/properties/EnchantmentPair.java
+@@ -0,0 +1,23 @@
++package io.papermc.paper.inventory.item.properties;
++
++import org.bukkit.enchantments.Enchantment;
++import org.jetbrains.annotations.NotNull;
++
++/**
++ * Represents a serialized enchantment with its level.
++ */
++public interface EnchantmentPair {
++
++    /**
++     * Gets the enchantment
++     * @return enchantment
++     */
++    @NotNull Enchantment enchantment();
++
++    /**
++     * Gets the level
++     * @return level
++     */
++    int level();
++
++}
+diff --git a/src/main/java/io/papermc/paper/inventory/item/properties/ItemProperties.java b/src/main/java/io/papermc/paper/inventory/item/properties/ItemProperties.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..ad5bf6a0c3389a7505827441ab9cb39c10d79759
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/inventory/item/properties/ItemProperties.java
+@@ -0,0 +1,128 @@
++package io.papermc.paper.inventory.item.properties;
++
++import com.destroystokyo.paper.Namespaced;
++import com.destroystokyo.paper.profile.PlayerProfile;
++import io.leangen.geantyref.TypeFactory;
++import io.papermc.paper.math.Position;
++import io.papermc.paper.property.MutablePropertyView;
++import io.papermc.paper.property.PropertyKey;
++import net.kyori.adventure.text.Component;
++import org.bukkit.Color;
++import org.bukkit.DyeColor;
++import org.bukkit.FireworkEffect;
++import org.bukkit.MusicInstrument;
++import org.bukkit.NamespacedKey;
++import org.bukkit.entity.Axolotl;
++import org.bukkit.inventory.ItemFlag;
++import org.bukkit.inventory.ItemStack;
++import org.bukkit.inventory.meta.BookMeta;
++import org.bukkit.inventory.meta.trim.ArmorTrim;
++import org.bukkit.potion.PotionData;
++import org.bukkit.potion.PotionEffect;
++import org.jetbrains.annotations.ApiStatus;
++
++import java.util.List;
++import java.util.Set;
++
++@ApiStatus.Experimental
++public interface ItemProperties {
```

Comment:
I don't think the benefit of less `public static final` is worth having a public interface that can be extended regardless of annotations.